pages/10.cookbook/06.tutorials/02.speaking-urls-for-flex-objects/docs.md

```diff
@@ -0,0 +1,166 @@
+
+[prism classes="language-md line-numbers"]
+---
+title: Single Place
+fx_directory: 'places'
+fx_title: 'title'
+fx_object: null
+---
+[/prism]
+
+### Template
+
+`place.html.twig` will look a bit like this:
+
+[prism classes="language-twig line-numbers"]
+{% extends 'partials/base.html.twig' %}
+
+{% set places = grav.get('flex').collection( header.fx_directory ) %}
+
+{% block content %}
+    <h1 class="headline">
+        {{ page.title }}
+    </h1>
+    <ul class="places__list">
+        {% for place in places %}
+            <li class="places__item">
+                <a href="{{ uri.path }}/{{ place.slug }}">
+                    {{ place.title }}
+                </a>
+            </li>
+        {% endfor %}
+    </ul>
+{% endblock %}
+[/prism]
+
+`place.html.twig` will look a bit like this:
+
+[prism classes="language-twig line-numbers"]
+{% extends 'partials/base.html.twig' %}
+
+{% set object = page.header.fx_object %}
+
+{% block content %}
+    <h1 class="headline">
+        {{ object.title }}
+    </h1>
+    <div class="textflow">
+        {{ object.description|markdown|raw }}
+    </div>
+{% endblock %}
+[/prism]
+
+### Routing
+
+In your themes `THEMENAME.php` add the following:
+
+[prism classes="language-php line-numbers"]
+use Grav\Common\Uri;
+
+public static function getSubscribedEvents(): array
+{
+    return [
+        'onPagesInitialized'       => ['onPagesInitialized', 0],
+    ];
+}
+
+public function onPagesInitialized(): void
+{
+    $current = Uri::getCurrentRoute();
+    $route = $current->getRoute();
+    $this->customRouting($route);
+}
+
+protected function customRouting( string $route ): void
+{
+    $normalized = trim($route, '/');
+    if (!$normalized) {
+        return;
+    }
+
+    // split the base from the slug
+    $parts = explode('/', $normalized, 2);
+    $path = array_shift($parts);
+    $key = array_shift($parts);
+
+    // get the reference to the details page corpus
+    $parent = $this->grav['pages']->find( '/' . $path );
+    $entity_page = $parent->header()->fx_entity_page ?? null;
+
+    // if we have a slug and a valid page to fill
+    if ( $entity_page && $key )
+    {
+        $entity_path = '/' . $path . '/' . $entity_page;
+        $this->addFlexPage( $route, $key, $entity_path ) ;
+    }
+}
+
+protected function addFlexPage( string $route, string $key, string $fx_page ): void
+{
+    /** @var Pages $pages */
+    $pages = $this->grav['pages'];
+    if ( $pages->find( $route ) )
+    {
+        /** @var Debugger $debugger */
+        $debugger = $this->grav['debugger'];
+        $debugger->addMessage("Page {$route} already exists, page cannot be added", 'error');
+        return;
+    }
+
+    // grab the single page corpus for later rerouting
+    $page = $pages->find( $fx_page );
+
+    // get the flex object to ensure it exists
+    $collection = $this->grav['flex']->getCollection( $page->header()->fx_directory );
+    $object = $collection->filterBy( ['slug'=> $key] )->first();
```

Comment:
those lines could be reduced to
`$object = $this->grav['flex']->getObject( $key, $page->header()->fx_directory, 'slug' );`
but i did not get it to work (returned null), so feel free to improve